### PR TITLE
Steam Oven, a few buffes to foods & some new stuffs

### PR DESCRIPTION
**Pull request by bioastroiner**

### CODE REVIEW

File: src/main/java/gregtechfoodoption/machines/GTFOTileEntities.java [hunk 1]
```diff
@@ -53,6 +55,7 @@ public static void init() {
         }
         BAKING_OVEN = registerMetaTileEntity(8516, new MetaTileEntityBakingOven(location("baking_oven")));
         ELECTRIC_BAKING_OVEN = registerMetaTileEntity(8517, new MetaTileEntityElectricBakingOven(location("electric_baking_oven")));
+        STEAM_BAKING_OVEN = registerMetaTileEntity(8544,new MetaTileEntitySteamBakingOven(location("steam_baking_oven"),GTFORecipeMaps.STEAM_BAKING_OVEN_RECIPE,10));
```
Comment (bruberu) [inline src/main/java/gregtechfoodoption/machines/GTFOTileEntities.java:58]: Formatting

File: src/main/java/gregtechfoodoption/machines/multiblock/MetaTileEntitySteamBakingOven.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.util.ResourceLocation;
+
+import javax.annotation.Nonnull;
+
+import static gregtech.api.unification.material.Materials.Steel;
+
+public class MetaTileEntitySteamBakingOven extends RecipeMapSteamMultiblockController {
+    public MetaTileEntitySteamBakingOven(ResourceLocation metaTileEntityId, RecipeMap<?> recipeMap, double conversionRate) {
+        super(metaTileEntityId, GTFORecipeMaps.STEAM_BAKING_OVEN_RECIPE, CONVERSION_RATE);
```
Comment (bruberu) [inline src/main/java/gregtechfoodoption/machines/multiblock/MetaTileEntitySteamBakingOven.java:27]: Although the recipe map does use temperature, this does not use it in the slightest.

File: src/main/java/gregtechfoodoption/recipe/GTFORecipeMaps.java [hunk 3]
```diff
@@ -45,16 +45,28 @@
             .setSlotOverlay(false, false, true, GuiTextures.FURNACE_OVERLAY_1)
             .setProgressBar(GuiTextures.PROGRESS_BAR_ARROW_MULTIPLE, ProgressWidget.MoveType.HORIZONTAL);
 
+    @ZenProperty
+    public static final RecipeMap<ElectricBakingOvenRecipeBuilder> STEAM_BAKING_OVEN_RECIPE = new ElectricBakingOvenRecipeMap<>("steam_baking_oven", new ElectricBakingOvenRecipeBuilder())
```
Comment (bruberu) [inline src/main/java/gregtechfoodoption/recipe/GTFORecipeMaps.java:49]: Personally, I don't think this recipe map is actually needed, given that it only is the regular baking oven map with durations halved and steam input, which can be mirrored by other means. The temperature is also borrowed from the ElectricBakingOvenRecipeBuilder, which adds unneeded information.

File: src/main/java/gregtechfoodoption/recipe/chain/BreadsChain.java [hunk 4]
```diff
@@ -243,7 +237,10 @@ public static void init() {
...
         ModHandler.addShapelessRecipe("gtfo_sandwich_bacon", SANDWICH_BACON.getStackForm(), PRESLICED_BREAD, BACON, BACON, BACON, BACON, BACON, BACON, BACON, BACON);
-        ModHandler.addShapelessRecipe("gtfo_sandwich_steak", SANDWICH_STEAK.getStackForm(), PRESLICED_BREAD, GTFOMaterialHandler.MeatIngot.getItemStack(), CHEDDAR_SLICE, CHEDDAR_SLICE);
+        ModHandler.addShapelessRecipe("gtfo_sandwich_steak_from_meat", SANDWICH_STEAK.getStackForm(), PRESLICED_BREAD, GTFOMaterialHandler.MeatIngot.getItemStack());
+        ModHandler.addShapelessRecipe("gtfo_sandwich_steak_from_steak", SANDWICH_STEAK.getStackForm(), PRESLICED_BREAD, new ItemStack(Items.COOKED_BEEF));
+        ModHandler.addShapelessRecipe("gtfo_large_sandwich_steak_from_steak", SANDWICH_LARGE_STEAK.getStackForm(), PRESLICED_BAGUETTE, new ItemStack(Items.COOKED_BEEF), new ItemStack(Items.COOKED_BEEF), new ItemStack(Items.COOKED_BEEF));
+        ModHandler.addShapelessRecipe("gtfo_large_sandwich_steak_from_meat", SANDWICH_LARGE_STEAK.getStackForm(), PRESLICED_BAGUETTE, GTFOMaterialHandler.MeatIngot.getItemStack(), GTFOMaterialHandler.MeatIngot.getItemStack(), GTFOMaterialHandler.MeatIngot.getItemStack());
```
Comment (bruberu) [inline src/main/java/gregtechfoodoption/recipe/chain/BreadsChain.java:243]: Large sandwiches are gated behind the cuisine assembler in order to make it more desirable, so I would prefer not to have these recipes.

File: src/main/resources/assets/gregtechfoodoption/lang/en_us.lang [hunk 5]
```diff
@@ -254,8 +255,11 @@ fluid.gtfo_starch_filled_water=Starch-Filled Water
 fluid.gtfo_hot_frying_oil=Hot Frying Oil
 
 gregtechfoodoption.machine.baking_oven.name=Primitive Baking Oven
+gregtechfoodoption.machine.steam_baking_oven.name=Steam Baking Oven
+gregtechfoodoption.machine.steam_baking_oven.tooltip=The Inner Casings can be either Tempted Glass or Casings but not IO parts! works double as fast as normal cooking oven and half the speed of electric version.
```
Comment (bruberu) [inline src/main/resources/assets/gregtechfoodoption/lang/en_us.lang:259]: Capitalization, and it should be "twice as fast".

**Review by bruberu — CHANGES_REQUESTED** [review]
There are a few things that I would like to be addressed before this goes through. Also, for the reinforced adobe bricks, the texture oculd be improved, as it looks a little too technological for a baking oven at the moment.

File: src/main/java/gregtechfoodoption/recipe/GTFORecipeMaps.java (hunk 3 above)
Comment (bioastroiner) [reply src/main/java/gregtechfoodoption/recipe/GTFORecipeMaps.java]: Its difference with normal Baking oven recipe map is that normal baking takes 2 input and fuel while steam oven acts like electric version without temperature value, so it's something in between

**Review by bruberu — APPROVED** [review]

**Pull request merged**